packages/go/scenepack/manager.go

```
@@ -0,0 +1,472 @@
+	}
+	stg := filepath.Join(m.root, "staging", id)
+	if _, err := os.Stat(stg); err != nil {
+		return fmt.Errorf("scenepack: staging/%s 不在（先 StageInstall）: %w", id, err)
+	}
+	cur := filepath.Join(m.root, "registry", id)
+	prev := cur + prevSuffix
+	// rollback 错误路径统一收口：旧版复位（若曾让位）+ staging 全弃（随时可弃面）
+	// ——优雅失败 0 残留（T16-G0-02）；崩溃（panic）不走此路，归 Recover 收敛。
+	rollback := func() {
+		if _, err := os.Stat(prev); err == nil {
+			os.Rename(prev, cur)
```

golangci-lint (2.12.2)

[error] Error return value of `os.Rename` is not checked

(errcheck)